mitosheet/mitosheet/step_performers/concat.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+from copy import copy
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+
+import pandas as pd
+from mitosheet.state import DATAFRAME_SOURCE_CONCAT, State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.transpiler.transpile_utils import (
+    column_header_list_to_transpiled_code, column_header_to_transpiled_code)
+from mitosheet.types import ColumnHeader, ColumnID
+
+class ConcatStepPerformer(StepPerformer):
+    """
+    Allows you to concatenage two or more dataframes together.
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'concat'
+    
+    @classmethod
+    def step_display_name(cls) -> str:
+        return 'Concatenated Dataframes'
+
+    @classmethod
+    def step_event_type(cls) -> str:
+        return 'concat_edit'
+
+    @classmethod
+    def saturate(cls, prev_state: State, params: Dict[str, Any]) -> Dict[str, Any]:
+        return params
+
+    @classmethod
+    def execute( # type: ignore
+        cls,
+        prev_state: State,
+        join: str, # inner | outter
+        ignore_index: bool,
+        sheet_indexes: List[int],
+        **params
+    ) -> Tuple[State, Optional[Dict[str, Any]]]:
+
+        post_state = copy(prev_state)
+
+        to_concat = [post_state.dfs[sheet_index] for sheet_index in sheet_indexes]
+
+        pandas_start_time = perf_counter()
+        new_df = pd.concat(to_concat, join=join, ignore_index=ignore_index)
+        pandas_processing_time = perf_counter() - pandas_start_time
+
+        # Add this dataframe to the new post state
+        post_state.add_df_to_state(new_df, DATAFRAME_SOURCE_CONCAT)
+
+        return post_state, {
+            'pandas_processing_time': pandas_processing_time
+        }
+
+    @classmethod
+    def transpile( # type: ignore
+        cls,
+        prev_state: State,
+        post_state: State,
+        execution_data: Optional[Dict[str, Any]],
+        join: str, # inner | outter
+        ignore_index: bool,
+        sheet_indexes: List[int]
+    ) -> List[str]:
+
+        df_names_to_concat = [post_state.df_names[sheet_index] for sheet_index in sheet_indexes]
+        df_new_name = post_state.df_names[len(post_state.dfs) - 1]
```

Comment:
I wonder how often this generates code that overwrites a previous dataframe that they've created. This is another place we have no visibility and so I'd be down to spend the 10 minutes to create a helper function that makes sure this is a unique df name.

All we need to do is check the df_names and make sure that the name we're generating doesn't exist and then increment it one more time if it does exist, right?

Reply:
We have this func already, I just had to use it! Added it to the add_df utility, aka now this isn't an issue.